nixos/tests/networking.nix

```diff
@@ -155,38 +155,38 @@ let
       };
       testScript = { ... }:
         ''
-          startAll;
+          start_all()
 
-          $client->waitForUnit("network.target");
-          $router->waitForUnit("network-online.target");
+          client.wait_for_unit("network.target")
+          router.wait_for_unit("network-online.target")
 
           # Wait until we have an ip address on each interface
-          $client->waitUntilSucceeds("ip addr show dev eth1 | grep -q '192.168.1'");
-          $client->waitUntilSucceeds("ip addr show dev eth1 | grep -q 'fd00:1234:5678:1:'");
-          $client->waitUntilSucceeds("ip addr show dev eth2 | grep -q '192.168.2'");
-          $client->waitUntilSucceeds("ip addr show dev eth2 | grep -q 'fd00:1234:5678:2:'");
-
-          # Test vlan 1
-          $client->waitUntilSucceeds("ping -c 1 192.168.1.1");
-          $client->waitUntilSucceeds("ping -c 1 192.168.1.2");
-          $client->waitUntilSucceeds("ping -c 1 fd00:1234:5678:1::1");
-          $client->waitUntilSucceeds("ping -c 1 fd00:1234:5678:1::2");
-
-          $router->waitUntilSucceeds("ping -c 1 192.168.1.1");
-          $router->waitUntilSucceeds("ping -c 1 192.168.1.2");
-          $router->waitUntilSucceeds("ping -c 1 fd00:1234:5678:1::1");
-          $router->waitUntilSucceeds("ping -c 1 fd00:1234:5678:1::2");
-
-          # Test vlan 2
-          $client->waitUntilSucceeds("ping -c 1 192.168.2.1");
-          $client->waitUntilSucceeds("ping -c 1 192.168.2.2");
-          $client->waitUntilSucceeds("ping -c 1 fd00:1234:5678:2::1");
-          $client->waitUntilSucceeds("ping -c 1 fd00:1234:5678:2::2");
-
-          $router->waitUntilSucceeds("ping -c 1 192.168.2.1");
-          $router->waitUntilSucceeds("ping -c 1 192.168.2.2");
-          $router->waitUntilSucceeds("ping -c 1 fd00:1234:5678:2::1");
-          $router->waitUntilSucceeds("ping -c 1 fd00:1234:5678:2::2");
+          client.wait_until_succeeds("ip addr show dev eth1 | grep -q '192.168.1'")
+          client.wait_until_succeeds("ip addr show dev eth1 | grep -q 'fd00:1234:5678:1:'")
+          client.wait_until_succeeds("ip addr show dev eth2 | grep -q '192.168.2'")
+          client.wait_until_succeeds("ip addr show dev eth2 | grep -q 'fd00:1234:5678:2:'")
```

Comment:
the output could again be stored in one variable and then be checked with multiple asserts

Reply:
Not that straight forward with `wait_until_succeeds`.
We would have to re-implement the 'waiting' to group multiple tests here.
I'm not sure if that is desirable.

Reply:
You are right. i typically don't recommend this on wait_until_succeeds calls, but i did not notice that here. Forget the suggestion.

Reply:
One suggestion I could look into later:
Add a method `wait_until_<complete?>` that returns as context manager, so something like this is possible:
`python
with machine.wait_until_complete("ip addr show dev eth2") as result:
    assert  '192.168.2' in result
    assert 'fd00:1234:5678:2:' in result
`

Reply:
That looks interesting. That i suggested this on `wait_until_complete` code was my mistake (i am usually suggesting this on `succeed` calls for the reasons you state). Building extra-library is of course out of scope for this PR. Feel free to ignore.

Reply:
Turns out that my idea, although interesting, is somewhat [impossible](https://mail.python.org/pipermail/python-ideas/2013-May/020633.html).
I'll think about a different approach and send a new PR for a library function.